zebrad/src/commands/tests.rs

```diff
@@ -0,0 +1,41 @@
+use clap::Parser;
+
+use crate::commands::ZebradCmd;
+
+use super::EntryPoint;
+
+#[test]
+fn args_with_subcommand_pass_through() {
+    let test_cases = [
+        (true, false, vec!["zebrad"]),
+        (true, true, vec!["zebrad", "-v"]),
+        (true, true, vec!["zebrad", "--verbose"]),
+        (true, false, vec!["zebrad", "-h"]),
+        (true, false, vec!["zebrad", "--help"]),
+        (true, false, vec!["zebrad", "start"]),
+        (true, true, vec!["zebrad", "-v", "start"]),
+        (true, false, vec!["zebrad", "warn"]),
+        (true, false, vec!["zebrad", "start", "warn"]),
+        (false, false, vec!["zebrad", "help", "warn"]),
+    ];
+
+    for (should_be_start, should_be_verbose, args) in test_cases {
+        let args = EntryPoint::process_cli_args(args.iter().map(Into::into).collect());
+        let args =
+            EntryPoint::try_parse_from(args).expect("hardcoded args should parse successfully");
+
+        assert!(args.config.is_none(), "args.config should be none");
+        assert!(args.cmd.is_some(), "args.cmd should not be none");
+        assert_eq!(
+            args.verbose, should_be_verbose,
+            "process_cli_args should preserve top-level args"
+        );
+
+        if should_be_start {
+            assert!(
+                matches!(args.cmd(), ZebradCmd::Start(_)),
+                "args.cmd should be start"
```

Comment:
Can you also test that the start command is not used when this is false?

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/6801/commits/6b553d290759d64255389c43c1dfb6698313b28b.